File: bindings/go/binding_test.go
package tree_sitter_typescript_test

import (
	"testing"

	tree_sitter "github.com/tree-sitter/go-tree-sitter"
	tree_sitter_typescript "github.com/tree-sitter/tree-sitter-typescript/bindings/go"
)

func TestCanLoadTypescriptGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_typescript.LanguageTypescript())
	if language == nil {
		t.Errorf("Error loading TypeScript grammar")
	}
}

func TestCanLoadTSXGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_typescript.LanguageTSX())
	if language == nil {
		t.Errorf("Error loading TSX grammar")
	}
}

File: bindings/node/binding_test.js
const assert = require('node:assert');
const { test } = require('node:test');

const Parser = require('tree-sitter');

test('can load TypeScript grammar', () => {
  const parser = new Parser();
  assert.doesNotThrow(() => parser.setLanguage(require('./typescript')));
});

test('can load TSX grammar', () => {
  const parser = new Parser();
  assert.doesNotThrow(() => parser.setLanguage(require('./tsx')));
});
